src/NEPTypes.jl

```diff
@@ -102,19 +111,31 @@ All of the matrices ``A_0,...`` are of size ``n×n``
 and ``f_i`` are a functions. The  functions ``f_i`` must be defined
 for matrices in the standard matrix function sense.
 """
-    struct SPMF_NEP <: AbstractSPMF
+
+# Logic behind Ftype:
+#  The eltype(F(λ))=promote_type(eltype(λ),Ftype)
+
+    struct SPMF_NEP_dense{T,Ftype}  <: AbstractSPMF{T}  # T should be a AbstractMatrix
```

Comment:
If `T` has to be an `AbstractMatrix`, perhaps we could do `struct SPMF_NEP_dense{FType, T<:AbstractMatrix}`?

Reply:
You're right. Updated. I kept order of the params.